src/Microsoft.Health.Operations.Functions/Management/PurgeOrchestrationInstanceHistory.cs

```diff
@@ -65,18 +69,30 @@ public async Task Run([TimerTrigger(PurgeFrequencyVariable)] TimerInfo myTimer,
         }
 
         DateTimeOffset end = Clock.UtcNow - _minimumAge;
-        log.LogInformation("Purging all orchestration instances with status in {{{Statuses}}} that started before '{End}'.",
+        log.LogInformation("Purging all orchestration instances with status in {{{Statuses}}} that started before '{End}' and not in {{{ListofInstanceToSkipPurging}}}.",
             string.Join(", ", _statuses),
-            end);
+            end,
+            string.Join(", ", _instancesToSkipPurging));
 
-        PurgeHistoryResult result = await client.PurgeInstanceHistoryAsync(DateTime.MinValue, end.UtcDateTime, _statuses);
+        OrchestrationStatusQueryCondition condition = new OrchestrationStatusQueryCondition
+        {
+            CreatedTimeFrom = DateTime.MinValue,
+            CreatedTimeTo = end.UtcDateTime,
+            RuntimeStatus = _statuses
+        };
+
+        var instances = await client.ListInstancesAsync(condition, CancellationToken.None);
+
+        var instancesToPurge = instances.DurableOrchestrationState.Where(x => !_instancesToSkipPurging.Contains(x.InstanceId, StringComparer.OrdinalIgnoreCase));
 
-        if (result.InstancesDeleted > 0)
+        foreach(var instance in instancesToPurge)
         {
-            log.LogInformation("Deleted {Count} orchestration instances from storage.", result.InstancesDeleted);
+            var result = await client.PurgeInstanceHistoryAsync(instance.InstanceId);
+            log.LogInformation("Instance with {InstanceId} deleted from storage", result.InstancesDeleted);
```

Comment:
nit: `InstancesDeleted` is not the ID, is it?

Reply:
that was a mistake, updated